compiler/luci/pass/src/FoldShapePass.test.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/FoldShapePass.h"
+#include "PassTestGraphs.h"
+
+#include <luci/IR/CircleNodes.h>
+
+#include <gtest/gtest.h>
+#include <iostream>
+
+namespace
+{
+
+template <loco::DataType FromT, loco::DataType ToT>
+class FoldShapeGraph : public luci::ConstantFoldingAddTestGraph
+{
+public:
+  FoldShapeGraph(std::vector<uint32_t> input_shape)
+    : luci::ConstantFoldingAddTestGraph(input_shape, ToT)
+  {
+    _x = _g.nodes()->template create<luci::CircleConst>();
+    _x->name("x");
+    _x->dtype(FromT);
+    _x->rank(input_shape.size());
+    for (uint32_t i = 0; i < input_shape.size(); i++)
+      _x->dim(i).set(input_shape.at(i));
+    _x->shape_status(luci::ShapeStatus::VALID);
+
+    _shape = _g.nodes()->template create<luci::CircleShape>();
+    _shape->name("shape");
+    _shape->out_type(ToT);
+    _shape->input(_x);
+    _shape->shape({4});
+    _shape->rank(1);
+    _shape->dim(0).set(4);
+  }
+
+  loco::Node *createFoldedPattern() override { return _shape; }
+
+protected:
+  luci::CircleShape *_shape = nullptr;
+  luci::CircleConst *_x = nullptr;
+};
+
+/**
+ *  Graph that has a Shape Op
+ *
+ *    BEFORE
+ *
+ *            [Input]
+ *               |
+ *            [Shape]
+ *               |
+ *            [Output]
```

Comment:
This is not true. PTAL `ConstantFoldingTestGraph`

Reply:
May I ask you some more detail?

The description at the `ConstantFoldingTestGraph` is below

```cpp
 *      [Input]   [Folded pattern] (Implemented by child class)
 *           \    /
 *         [Connector] (Implemented by child class)
 *              |
 *           [Output]
```

But, IIUC, `Shape` operation takes one input, and this PR aims to fold that `Shape` op as a constant, am I right..?

Reply:
After offline talk, I've got what @jinevening intended!

(For note)
`ConstantFoldingTestGraph` make a graph for test constant folding
And the operation which is aimed to be folded is **NOT** a `Connector`. Instead, it is placed at `Folded pattern`

`ConstantFoldingAddTestGraph` is a `ConstantFoldingTestGraph` whose `Connector` is `Add` operation, and it is easy to use to make a graph for the test